packages/code/src/CopyButton/CopyButton.tsx

```diff
@@ -51,19 +56,45 @@ function CopyButton({ onCopy, contents }: CopyProps) {
     setCopied(true);
   };
 
+  const handleMouseEnter = () => {
+    setHovered(true);
+  };
+
+  const handleMouseLeave = () => {
+    setHovered(false);
+  };
+
+  const shouldClose = () => !hovered;
+
   return (
-    <IconButton
-      ref={setButtonNode}
+    <Tooltip
+      open={hovered}
+      setOpen={setHovered}
       darkMode={darkMode}
```

Comment:
This isn't part of your changes but, you should be able to remove `darkMode` from here since we pass `darkMode` to the provider in `code.tsx`

Reply:
done!